### PR TITLE
Change default theme to uids_base.

### LINKED ISSUE #1660: As a SiteNow developer, I can deploy a theme that uses UIDS with paragraphs (part two)
Following up from https://github.com/uiowa/uiowa/issues/1324
- [x] #1654 Account for background color options with paragraph bundles
- [x] Maybe covered by 1654, but dark gray bg with yellow links fails contrast ratio.
- ~~Support for page title `sub_title` (e.g. Person credentials, etc.)~~ _With the person credentials accounted for, there currently isn't another use case for sub_title right now_
- [x] https://github.com/uiowa/uiowa/issues/1617 Sidebar menus
- [x] #1662 wysiwyg stylesheets
- [x] https://github.com/uiowa/uids/issues/199, https://github.com/uiowa/uids/issues/194 IE11
- [x] https://github.com/uiowa/uiowa/issues/1619 Spacing above footer
- [x] https://github.com/uiowa/uiowa/issues/1633 Header adjustments
- [x] https://github.com/uiowa/uiowa/issues/1678 Preview header/actions
- [x] https://github.com/uiowa/uids/issues/123 UIDS Carousel
- [x] https://github.com/uiowa/uiowa/issues/1465 Events 
- [x] Featured Content
- [x] https://github.com/uiowa/uiowa/pull/1468 People
- [x] https://github.com/uiowa/uiowa/issues/1611 Migrate to new footer menus _Complete, waiting for deployment_
- [x] https://github.com/uiowa/uiowa/issues/1749 ighn.international.uiowa.edu
- [x] hr.uiowa.edu has been accounted for and will not break with this rollout. communications about adoption continue.
- [x] https://github.com/uiowa/uiowa/issues/1756 Reorganize sitenow directory within uids_base 
- [x] https://github.com/uiowa/uiowa/issues/1137 visual indication of unpublished content
- [x] #1811 Footer logo appears smaller in SiteNow sites
- [x] #1808 styling for nonlink <a> tags used for anchors
- [x] #1810 styling (primarily line-height for spacing) of text in `<div>` tags
- [x] https://github.com/uiowa/uiowa/issues/1812 Loss of default top padding on some paragraphs
- [x] https://github.com/uiowa/uiowa/issues/1801 Toggle menu issue.
- [x] https://github.com/uiowa/uiowa/issues/1815 Article display issues 
- [x] https://github.com/uiowa/uiowa/issues/1844 Article mobile display regression
- [x] #1817 Migrate uiowa_bar quick links
- [x] https://github.com/uiowa/uiowa/issues/1822 Restrict top links to two on non-homepage sites
- [x] #1820 'Additional Information' content from basic site settings
- ~~#1824 Site slogan, lockup unused/remains in basic site settings~~ _Based on current decisions, this can happen at a later date_
- [ ] #1775 a11y fixes
- [x] Communicate the upcoming change (alert, article, listserv, sprint demo) and provide docs about manual clean-up: https://github.com/uiowa/uiowa/issues/1401

There are possibly several other issues that need to be addressed before uids_base adoption happens for sites. Pick out random sites to pull down, and turn on uids_base locally. Create and relate issues to this one to be addressed.

### LINKED ISSUE #1749: UIDS_Base coverage for IGHN.International
Relates to #1660 

## Problem

ighn.international has some css overrides to account for their custom person content type and view. This needs to be revisited before adoption of uids_base.

## Proposed Solution

The current css overrides should only be loaded if the active theme is uiowa_bootstrap.

Create a yarn workspace for ighn to utilize UIDS components and extend/override to cover the custom functionality in IGHN.

Follow the same designs being done with the person full and teaser in uids_base. Remove the map header bg as it is "off-brand"

This should rollout in conjunction with the uids_base switch.

- [x] Set parent organization stuff in site settings/theme settings

### LINKED ISSUE #1817: Migrate uiowa_bar quick links
Relates to #1660 

As part of the move to uids_base and from uiowa_bar to uiowa_search, look into programmatically migrating links currently in quick links, possibly to the Top Links menu.

### PR DESCRIPTION
**Pull request by richardbporter**
Resolves #1660 
Resolves #1749
Resolves #1817 

### Notes
- ~~uiowa_bootstrap cannot be uninstalled yet since it is a dependency of other config entities.~~
- ~~Do we need seven installed? It is currently the maintenance theme.~~

### Testing
- The usual `git pull`, `composer install`, `blt frontend`.
- Run `blt dsa` for sites you are familiar with. Test everything!?
- Comment here with issues found.
- Check sites against this list: #1901 

## Tested:

- [x] virtualdance.studio.uiowa.edu - @joewhitsitt 
- [x] coronavirus.uiowa.edu @pyrello 
- [x] protostudios.uiowa.edu @cory-skeers 
- [x] hawkeyemarchingband.uiowa.edu @pyrello 
- [x] cnm.uiowa.edu @cory-skeers 
- [x] diversity.uiowa.edu @joewhitsitt 
- [x] itaccessibility.uiowa.edu @joewhitsitt
- [x] iti.uiowa.edu @joewhitsitt 
- [x] sitenow.uiowa.edu @richardbporter 
- [x] planning-and-development.fo.uiowa.edu @joewhitsitt 

## Other potential sites to test.
- ap-purchasing.fo.uiowa.edu
- bands.uiowa.edu
- businsvc.fo.uiowa.edu
- covidresponse.research.uiowa.edu
- dasblatt.sites.uiowa.edu
- honors.uiowa.edu
- iti.uiowa.edu
- miles.lab.uiowa.edu
- pac.org.uiowa.edu
- seru.uiowa.edu
- uibio.org.uiowa.edu
- vanallenprobes.centerforconferences.uiowa.edu

### CODE REVIEW

**Comment by bspeare** [comment]
Do you have any suggestions on complex sites to test besides https://hawkeyemarchingband.uiowa.edu/?

**Comment by richardbporter** [comment]
Probably anything with split config. IDK what the most complicate "standard" site is. I bet marchingband is probably a good one.

To find sites with split config, go here https://github.com/uiowa/uiowa/find/master and type config/.htaccess as that file gets exported through usual process. Look for for site URL in the path and ignore the env and features directories. Ex.
image

So sitenow.uiowa.edu is in that list. That's probably a good one.

**Comment by briand44** [comment]
> IDK what the most complicated "standard" site is.

@cory-skeers  might have some ideas. It seemed like we did programmatic migrations for some of the standard sites with a lot of content and Cory was the one that handled those.

**Comment by pyrello** [comment]
@richardbporter Probably better to remove uiowa_bootstrap in a follow-up issue anyways to avoid WSOD issues in the time between code update and config update.

**Comment by cory-skeers** [comment]
Probably the most complicated 'standard' site, content-wise, would be cnm.uiowa.edu. It's a large site, with a lot of styling done in text fields. It doesn't get too complex in using different paragraph types, though.

**Comment by richardbporter** [comment]
>Probably better to remove uiowa_bootstrap in a follow-up issue anyways to avoid WSOD issues in the time between code update and config update.

By uninstall I mean just disable the extension and not remove the theme yet. That shouldn't cause any downtime since it would happen per-site.

**Review by richardbporter — COMMENTED** [review]
Default site front page image seems blurry after this change. I'm guessing the breakpoints are different.

**Comment by pyrello** [comment]
Checked coronavirus.uiowa.edu and didn't find any issues other than #1811.

**Comment by pyrello** [comment]
@joewhitsitt I think in that update hook, we may want to also consider setting sites to use the header style with the site name inline with the IOWA bar, unless they have a org parent set. It seems that the default that has already been set on most sites is for the site name to show below the IOWA bar.

**Comment by richardbporter** [comment]
I'm testing sitenow.uiowa.edu which has some bootstrap wonkiness going on.

File: config/default/menu_breadcrumb.settings.yml [hunk 1]
```diff
@@ -83,4 +68,3 @@ _core:
 front_title: 0
 stop_on_first_match: false
 derived_active_trail: false
-exclude_disabled_menu_items: 1
```
Comment (richardbporter) [inline config/default/menu_breadcrumb.settings.yml]: This doesn't look right. I believe we just added this in an update.

**Comment by richardbporter** [comment]
I've merged in some uiowa_alerts changes to avoid having to run another large database update for all sites. Issues are linked in commits.

**Comment by cory-skeers** [comment]
Updated with changes to resolve #1817 . SiteNow site which uses Quick LInks will migrate these into Top Links.

**Comment by joewhitsitt** [comment]
trying to update branch locally. running through a couple site syncs to see if my merges worked before pushing up

**Review by richardbporter — COMMENTED** [review]
I re-exported the settings with the desired option. It did re-export top links config again. I don't know if that is problematic.

**Comment by richardbporter** [comment]
I'm confused why update hooks exist in master related to the menu changes but https://github.com/uiowa/uiowa/pull/1755 has not been merged yet.

File: config/default/menu_breadcrumb.settings.yml (hunk 1 above)
Comment (richardbporter) [reply config/default/menu_breadcrumb.settings.yml]: e6ae5b2e07d27a4142bf11ea8ecfe536cadedd17

**Comment by joewhitsitt** [comment]
Sean had merged in for testing, other PRs related to 1660 with the idea that they were far enough along to eventually merge and become part of this PR as master. Footer menus, part of 1660, is just waiting for the right deployment window.

**Comment by richardbporter** [comment]
I was going to merge master in and resolve the conflicts but was unclear what the desired state of the paragraphs SASS was. We'll need to reorder the update hooks since sitenow_update_8020 has already run.

**Comment by joewhitsitt** [comment]
For paragraphs, I believe master should win-out. Most styling work happened outside this PR and was merged in.

**Comment by richardbporter** [comment]
Testing out new site installs via `blt drupal:install` and noticed that the navigation toggle is present until saving theme settings again. I'm not sure what is controlling that - anyone know?

Screen Shot 2020-06-29 at 2 26 23 PM

**Comment by joewhitsitt** [comment]
sitenow_update_8022 is supposed solve that by recreating the same theme settings logic

**Comment by richardbporter** [comment]
> sitenow_update_8022 is supposed solve that by recreating the same theme settings logic

Yes but that doesn't impact newly created sites and it seems like the default config is reflective of what that update hook does, unless I'm missing something.

https://github.com/uiowa/uiowa/blob/sitenow-uids/config/default/uids_base.settings.yml

**Comment by joewhitsitt** [comment]
Ah. I see. I didn't think about new sites. Sounds like the default config for these blocks need the same update as what is happening in the update hook. ~~I can address this later if nobody else gets to it first.~~ a1eea21

**Comment by richardbporter** [comment]
In the design meeting, I expressed confusion over the plan for uiowa_footer, specifically the additional information in site settings. Currently, there is an error thrown on the site settings admin page. Seems like there are a couple issues related to this: #1824 and #1825.

**Comment by richardbporter** [comment]
Did anyone test hawkeyemarchingband.uiowa.edu? Notice it hasn't been checked.

**Comment by pyrello** [comment]
@richardbporter I did test it, but I haven't looked at it for over a week. Most of the issues I could find at the time have been addressed.

**Comment by pyrello** [comment]
This is currently blocked by #1847. The plan is to get that merged in the next couple days.

**Comment by joewhitsitt** [comment]
is this a concern, from the default site?

```
>  [error]  File at 'public://2019-02/2018_09_01-Iowa v N. Illinois-tschoon-978_0.jpg' does not exist (plugin: imagemagick_identify) (method: loadMetadataFromFile) 
>  [error]  File at 'public://2019-02/2018_09_01-Iowa v N. Illinois-tschoon-978_0.jpg' does not exist (plugin: imagemagick_identify) (method: loadMetadataFromFile) 
```

**Comment by cory-skeers** [comment]
The file errors are part of the image_background to image media type consolidation, and should only come up locally. Shouldn't come up if you rsync first, and doesn't impact testing post-import if there's a stage-proxy (the media entity will still reference the correct file).

**Comment by pyrello** [comment]
Here is the list of additional testing requirements that came along from the UIDS 3.0 update: #1901

**Comment by richardbporter** [comment]
Looks like a0db7ae accidentally re-enabled uiowa_bootstrap and seven in default config and removed it from HR split.

**Comment by joewhitsitt** [comment]
The coronavirus exception for header type will only work in PROD unless you change the update hook to look for the env domain (e.g. coronavirus.local.drupal.uiowa.edu)

**Comment by richardbporter** [comment]
Maybe we should be using the D8 equivalent of conf_path if it exists. Sounds like it might be the site.path service: https://www.drupal.org/project/drupal/issues/2384675

**Comment by richardbporter** [comment]
Just noticed this during default site sync:
```
 [warning] Entity view display 'media.image.edgy': Component 'field_media_image' was disabled because its settings depend on removed dependencies.
 [warning] Entity view display 'media.image.sitenow_16_9': Component 'field_media_image' was disabled because its settings depend on removed dependencies.
```

**Comment by pyrello** [comment]
@richardbporter I am seeing those warnings too, but when I log into the site afterwards, it looks like the entity_view_display for edgy and sitenow_16_9 is there and set as I would expect.

**Comment by pyrello** [comment]
The condition for the coronavirus exception was backwards. I fixed it in a subsequent commit.

**Comment by richardbporter** [comment]
I synced hawkeyemarchingband.uiowa.edu and noticed the /drumhawk page is borked.
image

**Comment by pyrello** [comment]
@richardbporter I'm looking into this.

**Comment by cory-skeers** [comment]
I think the image borkage is coming from the ordering of the image breakpoint ids in the responsive_image.styles--haven't looked into it directly yet, but I ran into a similar problem before where it would try to load the smallest version of an image for the largest area.

**Comment by pyrello** [comment]
@bspeare is taking at look at the layout issue on that page. This looks to be a regression since we just solved this a week or two ago when we switched to using layouts for managing page display.

**Comment by pyrello** [comment]
Found another small issue. This is on the Coronavirus travel page:
Screen Shot 2020-07-10 at 12 23 37 PM
Original:
Screen Shot 2020-07-10 at 12 24 27 PM

If you are looking at this locally, here is the URL: https://coronavirus.local.drupal.uiowa.edu/more-information/travel

**Comment by pyrello** [comment]
For the sake of not complicating this release by having to create an update to UIDS, I have added a temporary fix to UIDS Base for the alert issue I posted above. But I think ultimately, it should be handled in UIDS.

**Comment by joewhitsitt** [comment]
I had created an issue for the alert padding: https://github.com/uiowa/uiowa/issues/1933, I just pushed a fix for the media issue. pulling down some sites again for testing

**Comment by pyrello** [comment]
I moved the alert issue to UIDS: #280.

**Comment by pyrello** [comment]
I have been reviewing the coronavirus site and it looks good!

**Comment by bspeare** [comment]
The hawkeyemarchingband issue is being caused by the Slick slider.   I don't think this two column sidebar layout was ever tested with the Slick implementation.   I am looking at it now and trying to fix it.

**Comment by richardbporter** [comment]
That did it. Thanks. 👍

**Comment by richardbporter** [comment]
We should probably get this merged for additional testing on stage. Anyone have else in the works on this PR?

**Comment by pyrello** [comment]
I'm not working on anything. I was looking at HMB and it all looks good to me.

**Comment by bspeare** [comment]
The images are still borked on that HMB page at the mobile breakpoint.   Still working on that fix...

**Comment by pyrello** [comment]
I do not see what is being referenced.

**Comment by bspeare** [comment]
At the mobile breakpoint, on https://hawkeyemarchingband.local.drupal.uiowa.edu/drumhawk.

**Comment by bspeare** [comment]
It should be fixed now.

**Review by pyrello — APPROVED** [review]
Looks good, as far as I can tell!

**Comment by richardbporter** [comment]
Thanks all. Merging.

**Pull request merged**